parquet-avro/src/main/java/org/apache/parquet/avro/AvroIndexedRecordConverter.java

```diff
@@ -356,12 +356,16 @@ public void end() {
      * @param elementSchema the expected Schema for list elements
      * @return {@code true} if the repeatedType is the element schema
      */
-    static boolean isElementType(Type repeatedType, Schema elementSchema) {
+    static boolean isElementType(Type repeatedType, Schema elementSchema, String parentName) {
       if (repeatedType.isPrimitive() ||
           repeatedType.asGroupType().getFieldCount() > 1) {
         // The repeated type must be the element type because it is an invalid
         // synthetic wrapper (must be a group with one field).
         return true;
+      } else if (repeatedType.asGroupType().getFieldCount() == 1 &&
+          (repeatedType.getName().equals("array") ||
```

Comment:
Did you mean "only optional and *required* are allowed there in the 3-level structure"? This sounds good to me and is more general.